mitiq/shadows/post_processing.py

```diff
@@ -0,0 +1,141 @@
+from typing import Tuple, List, Union, Any
+
+import numpy as np
+
+PAULI_MAP = {"X": 0, "Y": 1, "Z": 2}
+
+
+def snapshot_state(
+    b_list: List[Union[int, float]], u_list: List[str]
+) -> np.ndarray[Any, Any]:
+    """
+    Implement a single snapshot state reconstruction,
+
+    Args:
+        b_list (array): The list of classical outcomes for the snapshot.
+        u_list (array): Indices for the applied Pauli measurement.
+
+    Returns:
+        Numpy array with the reconstructed snapshot.
```

Comment:
Maybe something like this? WDYT?
```suggestion
    Returns:
        Reconstructed quantum state in density operator form.
```